pytrips/structures/tripstype.py

```diff
@@ -121,13 +121,27 @@ def __lt__(self, other):
                 raise NotImplemented
         return other.subsumes(self)
 
+    def __le__(self, other):
+        if type(other) is str:
+            other = self.__ont[other]
+            if type(other) is not TripsType:
+                raise NotImplemented
+        return other.subsumes(self) or self.name == other.name
```

Comment:
Flip the order of checks -> equality first then subsumes.  Equality check is constant time, subsumes is not.